### PR TITLE
Rules set to match GNAT check

### PR DESCRIPTION
**Pull request by leogermond**
The full GNAT check set of rules is available at https://docs.adacore.com/live/wave/asis/html/gnatcheck_rm/gnatcheck_rm/predefined_rules.html

This set of commits aims to implement some of them in LKQL_checker.

+ Abort_Statements
+ Abstract_Types
+ Blocks
+ Function_Style_Procedures 
+ Tests for the GOTO_Statements rule
+ Add a "Checks" Ada project to run basic rules tests on, with an equivalent `coding_standard.rules` set for gnatcheck

### CODE REVIEW

**Comment by raph-amiard** [comment]
Hello Léo!

Please add tests for each of those checks (take https://github.com/AdaCore/langkit-query-language/tree/master/testsuite/tests/checks/volatile_objects as an example, it should be pretty simple)

**Review by raph-amiard — CHANGES_REQUESTED** [review]
Another very important thing missing is tests (one per rule). Take inspiration from the existing checker tests, it should be pretty easy to figure out :)

File: lkql_checker/share/lkql/rules.json [hunk 1]
```diff
@@ -5,6 +5,12 @@
         "params": []
```
Comment (raph-amiard) [inline lkql_checker/share/lkql/rules.json:5]: Missing test in this commit

File: lkql_checker/share/lkql/rules.json [hunk 2]
```diff
@@ -17,6 +17,12 @@
...
```
Comment (raph-amiard) [inline lkql_checker/share/lkql/rules.json:17]: Again, missing test

File: lkql_checker/share/lkql/controlled_type_declarations.lkql [hunk 3]
```diff
@@ -0,0 +1,8 @@
+fun canonical_fully_qualified_name(t) =
```
Comment (raph-amiard) [inline lkql_checker/share/lkql/controlled_type_declarations.lkql:1]: Again, missing test

**Review by raph-amiard — CHANGES_REQUESTED** [review]

**Review by raph-amiard — DISMISSED** [review]

**Review by t-14 — APPROVED** [review]

**Review by raph-amiard — APPROVED** [review]

**Pull request merged**